tutorials/algorithms/04_vqd.ipynb

```diff
@@ -0,0 +1,292 @@
+    "\n",
+    "num_qubits = 2\n",
+    "ansatz = TwoLocal(3, rotation_blocks=[\"ry\", \"rz\"], entanglement_blocks=\"cz\", reps=1)\n",
+    "\n",
+    "optimizer = SLSQP()\n",
+    "ansatz.draw('mpl')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we define the required primitives for running `VQD`. This algorihtm requires 2 different primitive instances: one `Estimator` for computing the expectation values for the \"VQE part\" of the algorithm, and one `Sampler` passed along to the `StateFidelity` subroutine that will be used to compute the cost for higher energy states. In this example, we will use the `ComputeUncompute` fidelity method for this cost computation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.primitives import Sampler, Estimator\n",
+    "from qiskit.algorithms.state_fidelities import ComputeUncompute\n",
+    "\n",
+    "estimator = Estimator()\n",
+    "sampler = Sampler()\n",
+    "fidelity = ComputeUncompute(sampler)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In order to set up the VQD algorithm, it is important to define two additional inputs: the number of states to compute (`k`) and the `betas` defined in the original VQD paper. These `betas` balance the contribution of each overlap term to the cost function. In this example, we are going to set up the following values:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "k = 3\n",
+    "betas = [50, 50, 50]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we can run our VQD algorithm, but let's define a callback first to store intermediate values:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "counts = []\n",
+    "values = []\n",
+    "steps = []\n",
+    "\n",
+    "def callback(eval_count, params, value, meta, step):\n",
+    "    counts.append(eval_count)\n",
+    "    values.append(value)\n",
+    "    steps.append(step)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.algorithms.eigensolvers import VQD\n",
+    "\n",
+    "vqd = VQD(estimator, fidelity, ansatz, optimizer, k=k, betas=betas, callback=callback)\n",
+    "result = vqd.compute_eigenvalues(operator = H2_op)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see our 3 state energies as part of the VQD result:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[-1.8572746  -1.24458464 -0.88272226]\n"
```

Comment:
Are there any theoretical values to check whether these results are correct? That would help the users feel confident that everything works.

Reply:
I added the exact solution from the numpy eigensolver.